## Factores Cualitativos "AguasFeo"

|                |   |
:----------------|---|
| **Nombre**     David Alejandro Rangel Rodríguez|   |
| **Fecha**     9 de febrero del 2026 |   |
| **Expediente*756203* |   |

In [23]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

In [24]:
Agf = pd.read_excel('Aguasfeo.xlsx')
Agf

Tamaño Hielo    Sabor   Score
0  grande   sin    kahlúa    8.2
1  grande   con    kahlúa    9.0
2   chica   sin    kahlúa    7.0
3   chica   con    kahlúa    7.5
4  grande   con  horchata   10.0
5   chica   sin  horchata    6.0
6   chica   con  horchata    6.4
7  grande   sin  horchata    9.5

In [27]:
Agf.columns

Index(['Tamaño', 'Hielo', 'Sabor ', 'Score', 'Tamaño_bin', 'Hielo_bin'], dtype='object')

In [28]:
Agf['Tamaño_bin'] = Agf['Tamaño'].map({'grande': 1, 'chica': 0})
Agf['Hielo_bin'] = Agf['Hielo'].map({'con': 1, 'sin': 0})
Agf['Kahlúa'] = Agf['Sabor '].map({'kahlúa': 1, 'horchata': 0})
Agf['Horchata'] = Agf['Sabor '].map({'horchata': 1, 'kahlúa':0})
Agf

Tamaño Hielo    Sabor   Score  Tamaño_bin  Hielo_bin  Kahlúa  Horchata
0  grande   sin    kahlúa    8.2           1          0       1         0
1  grande   con    kahlúa    9.0           1          1       1         0
2   chica   sin    kahlúa    7.0           0          0       1         0
3   chica   con    kahlúa    7.5           0          1       1         0
4  grande   con  horchata   10.0           1          1       0         1
5   chica   sin  horchata    6.0           0          0       0         1
6   chica   con  horchata    6.4           0          1       0         1
7  grande   sin  horchata    9.5           1          0       0         1

In [33]:
import statsmodels.api as sm
X = Agf.drop('Score',axis = 1)
X = X.drop('Tamaño',axis = 1)
X = X.drop('Hielo',axis = 1)
X = X.drop('Sabor ',axis = 1)

y = Agf['Score']

X = sm.add_constant(X)

model = sm.OLS(y, X)
results = model.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                  Score   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Mon, 09 Feb 2026   Prob (F-statistic):             0.0473
Time:                        17:41:25   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.3000      0.320     13.417      0.0

C:\Users\range\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=8 observations were given.
  return hypotest_fun_in(*args, **kwds)


### Parte 2

In [34]:
Agf['Tamaño_bin'] = Agf['Tamaño'].map({'grande': -1, 'chica': 1})
Agf['Hielo_bin'] = Agf['Hielo'].map({'con': 1, 'sin': -1})
Agf['Kahlúa'] = Agf['Sabor '].map({'kahlúa': 1, 'horchata': 0})
Agf['Horchata'] = Agf['Sabor '].map({'horchata': 1, 'kahlúa':0})
Agf

Tamaño Hielo    Sabor   Score  Tamaño_bin  Hielo_bin  Kahlúa  Horchata
0  grande   sin    kahlúa    8.2          -1         -1       1         0
1  grande   con    kahlúa    9.0          -1          1       1         0
2   chica   sin    kahlúa    7.0           1         -1       1         0
3   chica   con    kahlúa    7.5           1          1       1         0
4  grande   con  horchata   10.0          -1          1       0         1
5   chica   sin  horchata    6.0           1         -1       0         1
6   chica   con  horchata    6.4           1          1       0         1
7  grande   sin  horchata    9.5          -1         -1       0         1

In [35]:
import statsmodels.api as sm
X = Agf.drop('Score',axis = 1)
X = X.drop('Tamaño',axis = 1)
X = X.drop('Hielo',axis = 1)
X = X.drop('Sabor ',axis = 1)

y = Agf['Score']

X = sm.add_constant(X)

model = sm.OLS(y, X)
results = model.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                  Score   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Mon, 09 Feb 2026   Prob (F-statistic):             0.0473
Time:                        17:48:05   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.3000      0.185     28.644      0.0

C:\Users\range\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=8 observations were given.
  return hypotest_fun_in(*args, **kwds)


### Parte 3 (Comparación Betas y observaciones)

Betas en la regresion con **numeros buleanos**:
*  $\beta_0$ = 4.3
*  $\beta_1$ = 2.45
*  $\beta_2$ = 0.55
*  $\beta_3$ = 2.125
*  $\beta_4$ = 2.175

Betas en la regresion con **numeros -1, 1**:
*  $\beta_0$ = 5.3
*  $\beta_1$ = -1.225
*  $\beta_2$ = 0.275
*  $\beta_3$ = 2.625
*  $\beta_4$ = 2.675

**Observaciones:**

El cambio en las betas ocurre principalmente por la escala y el punto de referencia de las variables. Al usar números booleanos (0 y 1), la diferencia entre categorías es de una unidad, por lo que la beta representa el cambio total al pasar de una opción a otra; en cambio, al usar −1 y 1, la distancia se duplica a dos unidades, lo que provoca que las betas de las pendientes se reduzcan a la mitad para compensar ese "paso" más largo. Además, el intercepto B0 se desplaza porque en el modelo booleano representa el valor base cuando todo es cero, mientras que en el modelo −1/1 se mueve hacia el promedio global de los datos. Finalmente, el cambio de signo en β1(de positivo a negativo) se debe por invertir la lógica de la variable: antes "grande" sumaba al ser el valor positivo (1) y ahora "chica" es la que tiene el valor positivo, invirtiendo el sentido de la relación.